### ISSUE TITLE
Unable to get release/version.json at web

### ISSUE THREAD

**Issue by wwng2333** [AdguardTeam/AdGuardHome#5408]
### Prerequisites

- [X] I have checked the [Wiki](https://github.com/AdguardTeam/AdGuardHome/wiki) and [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions) and found no answer

- [X] I have searched other issues and found no duplicates

- [X] I want to report a bug and not ask a question

### Operating system type

Linux, Other (please mention the version in the description)
OS: Ubuntu 20.04.5 LTS x86_64
Kernel: Linux 5.4.0-137-generic
### CPU architecture

AMD64

### Installation

GitHub releases or script from README

### Setup

On one machine

### AdGuard Home version

v0.107.22

#### What did you do?
F5 at dashboard, and it reports at corner said:
#### Expected result
Normal check update
#### Actual result
Error: control/version.json | getting version info from https://static.adtidy.org/adguardhome/release/version.json: updater: HTTP GET https://static.adtidy.org/adguardhome/release/version.json: Get "https://static.adtidy.org/adguardhome/release/version.json": couldn't lookup host: static.adtidy.org | 502
#### Screenshots (if applicable)
![image](https://user-images.githubusercontent.com/17147265/215068260-882b1bc0-dc71-472d-9423-55da772af8cf.png)
#### Additional information
My dns able to get both A and AAAA record , but the machine dont have IPv6 access, but the DNS upstream have, so i get the DNS record like :
```bash
root@hostname:~# q static.adtidy.org
static.adtidy.org. 1m54s CNAME 1625341327.rsc.cdn77.org.
1625341327.rsc.cdn77.org. 54s AAAA 2a02:6ea0:c800::7
1625341327.rsc.cdn77.org. 54s AAAA 2a02:6ea0:c800::6
1625341327.rsc.cdn77.org. 54s AAAA 2a02:6ea0:c800::8
static.adtidy.org. 1m49s CNAME 1625341327.rsc.cdn77.org.
static.adtidy.org. 1m53s CNAME 1625341327.rsc.cdn77.org.
static.adtidy.org. 1m48s CNAME 1625341327.rsc.cdn77.org.
static.adtidy.org. 2m0s CNAME 1625341327.rsc.cdn77.org.
static.adtidy.org. 1m52s CNAME 1625341327.rsc.cdn77.org.
1625341327.rsc.cdn77.org. 1m0s A 89.187.187.20
1625341327.rsc.cdn77.org. 1m0s A 89.187.187.12
1625341327.rsc.cdn77.org. 1m0s A 89.187.187.14
```
I try to download it via curl, and it works.
```bash
root@hostname:~# ALL_PROXY='' curl -v https://static.adtidy.org/adguardhome/release/version.json
*   Trying 89.187.187.12:443...
* TCP_NODELAY set
* Connected to static.adtidy.org (89.187.187.12) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/certs/ca-certificates.crt
  CApath: /etc/ssl/certs
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.3 (IN), TLS handshake, Server hello (2):
* TLSv1.3 (IN), TLS handshake, Encrypted Extensions (8):
* TLSv1.3 (IN), TLS handshake, Certificate (11):
* TLSv1.3 (IN), TLS handshake, CERT verify (15):
* TLSv1.3 (IN), TLS handshake, Finished (20):
* TLSv1.3 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.3 (OUT), TLS handshake, Finished (20):
* SSL connection using TLSv1.3 / TLS_AES_256_GCM_SHA384
* ALPN, server accepted to use h2
* Server certificate:
*  subject: CN=1625341327.rsc.cdn77.org
*  start date: Dec  9 10:07:29 2022 GMT
*  expire date: Mar  9 10:07:28 2023 GMT
*  subjectAltName: host "static.adtidy.org" matched cert's "static.adtidy.org"
*  issuer: C=US; O=Let's Encrypt; CN=R3
*  SSL certificate verify ok.
* Using HTTP2, server supports multi-use
* Connection state changed (HTTP/2 confirmed)
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* Using Stream ID: 1 (easy handle 0x55631832d2f0)
> GET /adguardhome/release/version.json HTTP/2
> Host: static.adtidy.org
> user-agent: curl/7.68.0
> accept: */*
> 
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* old SSL session ID is stale, removing
* Connection state changed (MAX_CONCURRENT_STREAMS == 128)!
< HTTP/2 200 
< date: Fri, 27 Jan 2023 10:52:19 GMT
< content-type: application/json; charset=utf-8
< last-modified: Thu, 19 Jan 2023 12:52:11 GMT
< vary: Accept-Encoding
< etag: W/"63c93cfb-b9f"
< access-control-allow-origin: *
< cf-cache-status: DYNAMIC
< cf-ray: 7900ab1caba1b383-PRG
< x-accel-expires: @1674828310
< server: CDN77-Turbo
< x-77-nzt: AVm7uwqJGEL/DQsAAA
< x-77-nzt-ray: 630bb91331f154bde3acd363904d2226
< x-cache: HIT
< x-age: 2829
< x-77-pop: losangelesUSCA
< x-77-cache: HIT
< 
{
  "version": "v0.107.22",
  "announcement": "AdGuard Home v0.107.22 is now available!",
  "announcement_url": "https://github.com/AdguardTeam/AdGuardHome/releases/tag/v0.107.22",
  "selfupdate_min_version": "0.0",

  "download_linux_armv5": "https://static.adtidy.org/adguardhome/release/AdGuardHome_linux_armv5.tar.gz",
  "download_darwin_arm64": "https://static.adtidy.org/adguardhome/release/AdGuardHome_darwin_arm64.zip",
  "download_linux_armv7": "https://static.adtidy.org/adguardhome/release/AdGuardHome_linux_armv7.tar.gz",
  "download_windows_arm64": "https://static.adtidy.org/adguardhome/release/AdGuardHome_windows_arm64.zip",
  "download_linux_mipsle_softfloat": "https://static.adtidy.org/adguardhome/release/AdGuardHome_linux_mipsle_softfloat.tar.gz",
  "download_linux_amd64": "https://static.adtidy.org/adguardhome/release/AdGuardHome_linux_amd64.tar.gz",
  "download_linux_mips64le_softfloat": "https://static.adtidy.org/adguardhome/release/AdGuardHome_linux_mips64le_softfloat.tar.gz",
  "download_linux_armv6": "https://static.adtidy.org/adguardhome/release/AdGuardHome_linux_armv6.tar.gz",
  "download_freebsd_386": "https://static.adtidy.org/adguardhome/release/AdGuardHome_freebsd_386.tar.gz",
  "download_linux_mips64_softfloat": "https://static.adtidy.org/adguardhome/release/AdGuardHome_linux_mips64_softfloat.tar.gz",
  "download_freebsd_armv7": "https://static.adtidy.org/adguardhome/release/AdGuardHome_freebsd_armv7.tar.gz",
  "download_freebsd_arm64": "https://static.adtidy.org/adguardhome/release/AdGuardHome_freebsd_arm64.tar.gz",
  "download_linux_arm64": "https://static.adtidy.org/adguardhome/release/AdGuardHome_linux_arm64.tar.gz",
  "download_windows_amd64": "https://static.adtidy.org/adguardhome/release/AdGuardHome_windows_amd64.zip",
  "download_linux_386": "https://static.adtidy.org/adguardhome/release/AdGuardHome_linux_386.tar.gz",
  "download_linux_ppc64le": "https://static.adtidy.org/adguardhome/release/AdGuardHome_linux_ppc64le.tar.gz",
  "download_windows_386": "https://static.adtidy.org/adguardhome/release/AdGuardHome_windows_386.zip",
  "download_freebsd_amd64": "https://static.adtidy.org/adguardhome/release/AdGuardHome_freebsd_amd64.tar.gz",
  "download_openbsd_amd64": "https://static.adtidy.org/adguardhome/release/AdGuardHome_openbsd_amd64.tar.gz",
  "download_darwin_amd64": "https://static.adtidy.org/adguardhome/release/AdGuardHome_darwin_amd64.zip",
  "download_freebsd_armv5": "https://static.adtidy.org/adguardhome/release/AdGuardHome_freebsd_armv5.tar.gz",
  "download_freebsd_armv6": "https://static.adtidy.org/adguardhome/release/AdGuardHome_freebsd_armv6.tar.gz",
  "download_frontend": "https://static.adtidy.org/adguardhome/release/AdGuardHome_frontend.tar.gz",
  "download_openbsd_arm64": "https://static.adtidy.org/adguardhome/release/AdGuardHome_openbsd_arm64.tar.gz",
  "download_linux_mips_softfloat": "https://static.adtidy.org/adguardhome/release/AdGuardHome_linux_mips_softfloat.tar.gz"
}
* Connection #0 to host static.adtidy.org left intact
```

**Comment by wwng2333** [comment]
It works at v0.107.21, and not work since i upgrade via web to v0.107.22

**Comment by wwng2333** [comment]
new error :
```
Error: control/version.json | getting version info from https://static.adtidy.org/adguardhome/release/version.json: updater: HTTP GET https://static.adtidy.org/adguardhome/release/version.json: Get "https://static.adtidy.org/adguardhome/release/version.json": couldn't dial to [2a02:6ea0:d600::2]:443: dial tcp [2a02:6ea0:d600::2]:443: connect: cannot assign requested address | 502
```

**Comment by Bmartin3k** [comment]
I'm getting this same error since updating.

**Comment by wwng2333** [comment]
Maybe need to setup a bootstrap dns for resolve static.adtidy.org

**Comment by wwng2333** [comment]
@chermy not same, but similar problem. #5373 is like 404, but i get 502 or even cant resolve the hostname here.

**Comment by Gooman-rus** [comment]
I have the same problem on v0.107.29.

Error response on POST requrst https://myhost/control/version.json:
`getting version info from https://static.adtidy.org/adguardhome/release/version.json: updater: HTTP GET https://static.adtidy.org/adguardhome/release/version.json: Get "https://static.adtidy.org/adguardhome/release/version.json": resolving "static.adtidy.org": all upstreams failed to respond: 4 errors: "returned: \"reading response from tls://1dot1dot1dot1.cloudflare-dns.com:853: read tcp 79.137.206.234:50962->1.0.0.1:853: read: connection reset by peer\", deferred: \"tls: failed to send closeNotify alert (but connection was closed anyway): write tcp 79.137.206.234:50962->1.0.0.1:853: write: broken pipe\"", "requesting https://cloudflare-dns.com:443/dns-query: Get \"https://cloudflare-dns.com:443/dns-query?dns=AAABAAABAAAAAAABBnN0YXRpYwZhZHRpZHkDb3JnAAAcAAEAACkIAAAAgAAAAA\": read tcp 79.137.206.234:39032->104.16.248.249:443: read: connection reset by peer", "requesting https://dns.google:443/dns-query: Get \"https://dns.google:443/dns-query?dns=AAABAAABAAAAAAABBnN0YXRpYwZhZHRpZHkDb3JnAAAcAAEAACkIAAAAgAAAAA\": read tcp 79.137.206.234:46810->8.8.4.4:443: read: connection reset by peer", "getting conn to tls://dns.google:853: connecting to dns.google: read tcp 79.137.206.234:60338->8.8.4.4:853: read: connection reset by peer"`

**Comment by ainar-g** [comment]
Apologies for a long time with no response.  The error suggests that your upstreams cannot resolve the addresses for AdGuard servers hosting the updates, see #5976.  Please try using a different upstream configuration.
